packages/utils/test/mergeDefaultsWithFormData.test.ts

```diff
@@ -307,4 +307,94 @@ describe('mergeDefaultsWithFormData()', () => {
       expect(mergeDefaultsWithFormData(obj1, obj2)?.a).toBeInstanceOf(File);
     });
   });
+
+  describe('test with overrideFormDataWithDefaults set to `replace`', () => {
+    it('should return empty array even when no defaults', () => {
+      expect(mergeDefaultsWithFormData(undefined, [2], undefined, undefined, 'replace')).toEqual([]);
+    });
```

Comment:
Maybe I'm misunderstanding the feature, why does this return an empty array?

Reply:
That's because undefined will be treated as missing value for this, so replaced with proper default for arrays (empty). Do you think it's incorrect?

Reply:
My assumption was that `mergeDefaultsWithFormData` treats `undefined` computed defaults as there being no defaults, so I would think in this scenario we keep user data. But if it's the case that it's only called when there is some default value (even if it's computed to be `undefined`), then it makes sense to replace user data with some value like `[]`. So I think what you have is fine. We can refine the implementation over time too if necessary.